lib/src/core/Model.dart

```diff
@@ -29,12 +28,11 @@ class Model implements IModel
    *
    * -  Throws [MultitonErrorModelExists] if instance for this Multiton key instance has already been constructed.
    */
-  Model( String key )
-  {
-    if ( instanceMap[ key ] != null ) throw new MultitonErrorModelExists();
-    multitonKey = key;
-    instanceMap[ multitonKey ] = this;
-    proxyMap = new Map<String,IProxy>();
+  Model(String key) {
+    if (instanceMap.containsKey(key)) throw MultitonErrorModelExists();
+    _multitonKey = key;
```

Comment:
Change `_multitonKey` back to `multitonKey`.

Reply:
Setter added for every places near getter.